mtypes/iso8601.go

```diff
@@ -0,0 +1,65 @@
+package mtypes
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/mailgun/errors"
+)
+
+// ISO8601Time Mailgun uses ISO8601 format for timestamps in the API key API endpoints ('2011/10/13T18:02:00'), but
+// by default Go's JSON package uses another format when decoding/encoding timestamps. This goes against the documentation
+// at https://documentation.mailgun.com/docs/mailgun/api-reference#date-format but here we are.
```

Comment:
RFC822 time is a MIME format. It's not really supposed to be an API format, so we are OK with this.

Reply:
Not sure if the above is just a comment or a request for change.

Reply:
Yeah, please remove that:
```suggestion
// by default Go's JSON package uses another format when decoding/encoding timestamps.
```